src/hotspot/share/utilities/globalDefinitions_gcc.hpp

```diff
@@ -31,6 +31,8 @@
 // globally used constants & types, class (forward)
 // declarations and a few frequently used utility functions.
 
+#include <cstddef>
```

Comment:
This should be in globalDefinitions.hpp, not globalDefinitions_gcc.hpp.  There's no guarantee that the inclusion of <stddef.h> on other platforms is sufficient to provide the std-qualified name.  In fact, that's explicitly unspecified and deprecated behavior.  (Unqualified nullptr_t being provided by <stddef.h> is deprecated, BTW.)

Reply:
You're right. I was blind-sided by the fact that this was only was supposed to only be a problem with clang. I'm updating the patch.